sttp/client/src/main/scala/endpoints/sttp/client/Endpoints.scala

```diff
@@ -0,0 +1,112 @@
+package endpoints.sttp.client
+
+import java.net.URI
+
+import endpoints.algebra
+import endpoints.Tupler
+import com.softwaremill.sttp
+
+import scala.language.higherKinds
+
+/**
+  * An interpreter for [[endpoints.algebra.Endpoints]] that builds a client issuing requests using
+  * a sttp’s [[com.softwaremill.sttp.SttpBackend]].
+  *
+  * Doest not support streaming responses for now
+  *
+  * @param host     Base of the URL of the service that implements the endpoints (e.g. "http://foo.com")
+  * @param backend  The underlying backend to use
+  * @tparam R       The monad wrapping the response. It is defined by the backend
+  */
+class Endpoints[R[_]](host: String, val backend: sttp.SttpBackend[R, Nothing]) extends algebra.Endpoints with Urls with Methods {
+
+  /**
+    * A function that, given an `A` and a request model, returns an updated request
+    * containing additional headers
+    */
+  type RequestHeaders[A] = (A, sttp.Request[_, Nothing]) => sttp.Request[_, Nothing]
+
+  /** Does not modify the request */
+  lazy val emptyHeaders: RequestHeaders[Unit] = (_, request) => request
+
+  /**
+    * A function that takes an `A` information and returns a `sttp.Request`
+    */
+  type Request[A] = A => sttp.Request[_, Nothing]
+
+  /**
+    * A function that, given an `A` information and a `sttp.Request`, eventually returns a `sttp.Request`
+    */
+  type RequestEntity[A] = (A, sttp.Request[_, Nothing]) => sttp.Request[_, Nothing]
+
+  lazy val emptyRequest: RequestEntity[Unit] = { case (_, req) => req }
+
+  def request[A, B, C, AB](
+    method: Method, url: Url[A],
+    entity: RequestEntity[B], headers: RequestHeaders[C]
+  )(implicit tuplerAB: Tupler.Aux[A, B, AB], tuplerABC: Tupler[AB, C]): Request[tuplerABC.Out] =
+    (abc: tuplerABC.Out) => {
+      val (ab, c) = tuplerABC.unapply(abc)
+      val (a, b) = tuplerAB.unapply(ab)
+
+      val uri: sttp.Id[sttp.Uri] = sttp.Uri(new URI(s"${host}${url.encode(a)}"))
+      val reqId: sttp.Request[_, Nothing] = method(sttp.sttp.get(uri = uri))
+      entity(b, headers(c, reqId))
+    }
+
+  /**
+    * Trait that indicates how a response should be interpreted
+    */
+  trait SttpResponse[A] {
+    /**
+      * The type of the received body from the server
+      */
+    type RB
+
+    /**
+      * To read the response body
+      */
+    def responseAs: sttp.ResponseAs[RB, Nothing]
+
+    /**
+      * Function to validate the response (headers, code). This can also modify the type of the received body
+      */
+    def validateResponse(response: sttp.Response[RB]): Either[String, A]
```

Comment:
I’ve just had a look at sttp’s documentation and it seems that `R` is not just a `Monad` but even a `MonadError`, so I suggest that you return a `R[A]` here, instead of a `Either[String, A]`.